src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs

```diff
@@ -67,8 +68,14 @@ ILogger<SearchController> logger
         [HttpGet]
         public async Task<IActionResult> Search(SearchIndexViewModel viewModel, PagerSlimParameters pagerParameters)
         {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.QueryLuceneApi))
+            {
+                return this.ChallengeOrForbid();
```

Comment:
Move to line 79

Reply:
@TFleury not sure if it is correct, can you check?